### ISSUE TITLE
Unable to Delete Alerts or Set Alerts to False Positive

### ISSUE THREAD

**Issue by twright-0x1** [zaproxy/zaproxy#2593]
In ZAP 2.5.0, when I attempt to delete one or more alerts, the alerts disappear from the Alerts tree (in the Alerts tab), but still show up in any HTML reports I generate.

If I attempt to mark an alert as false positive, it will appear as such in the Alerts tree, but not in an HTML report (it will appear as a true positive in the report).

Trying things like refreshing the alerts tree, or exiting and restarting ZAP (after deleting or marking alerts) does not have any effect on this issue.

**Comment by thc202** [comment]
I was able to reproduce the issue (also with 2.4.x versions).

**Comment by twright-0x1** [comment]
Also, I now notice that sometimes alerts that have been deleted show up in the Alerts tree after a restart of ZAP (and a reload of the session in question).

**Comment by thc202** [comment]
Yeah, saw that too (some of the alerts are not being deleted from the session).

**Comment by twright-0x1** [comment]
Continuing to see this in version 2.7.

**This is a very significant problem for ZAP.**  I'm having to hand-edit HTML reports that ZAP generates in order to remove alerts that can't be deleted from the Alerts tree.  Plus, because the errant alerts don't go away from the session, when I return to the session some time later, I may not realize which alerts are legit.

**Comment by stijnvancrombrugge** [comment]
I'm facing the same issue in version 2.7. Any updates on this ?

**Comment by i4s-pserrano** [comment]
I have the same issue here! Version 2.7

**Comment by twright-0x1** [comment]
Continuing to see this in version 2.8.0.  Although sometimes deleting an alert seems to work; my experience is that most of the time, it does not.  Per my earlier note, this is a real hardship when working with ZAP.

**Comment by jjuuddyyjenny** [comment]
Have the same problem! Any updates will be highly appreciated.

**Comment by rh-id** [comment]
hi, after checking the source code. I wonder if this issue was caused by below:

is this class `org.parosproxy.paros.db.paros.ParosTableAlert` used to store the Alert records in the database.
if so why it doesn't have any `connection.commit()` is this the correct way to do in this project?

**Comment by thc202** [comment]
This is not a problem with the database but in how the alerts are displayed in the tree (it's possible that the UI does not show all alerts, so it might appear that all were deleted or marked FP). The session uses auto-commit.

**Comment by twright-0x1** [comment]
This remains an active problem and a significant hurdle when using ZAP (which I otherwise really appreciate using).

In order to provide HTML reports to clients, I end up having to manually edit ZAP's generated report markup to remove alerts that I attempted to delete from the alert tree.

Moreover, if I reload a session after closing it out, some portion of the alerts I deleted show back up in the alert tree.

**Comment by kingthorin** [comment]
> This remains an active problem

That's why it's still open....

**Comment by penny-1995** [comment]
If this is solved, the reporting can be greatly improved as more things will be under control.

**Comment by MrTheV** [comment]
I have the same issue with ZAP 2.10, though I did not have it previously with ZAP 2.9.0...
I have ZAP 2.10.0 and 2.9.0 installed on the same PC (Windows 10 x64).
I snapshoted the session under ZAP 2.10.0, closed it, opened the snapshot under ZAP 2.9.0 and... some of the deleted alerts are showing up in the alerts tree. If I then delete them and generate an HTML report from ZAP 2.9.10, it works as expected: deleted alert are not in the HTML report.
So far, I have then to keep ZAP 2.9.0 to sanitize the alerts list/tree and reports...
